src/oop/programs_with_multiple_classes/hard/q4/MainTest.java

```diff
@@ -0,0 +1,150 @@
+package oop.programs_with_multiple_classes.hard.q4;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.time.Year;
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private ObjectTest book;
+    private ObjectTest bookcase;
+
+    private final String bookLc = "book";
+    private final String bookUc = "Book";
+    private final String bookcaseLc = "bookcase";
+    private final String bookcaseUc = "Bookcase";
+    private final String year = "year";
+    private final String type = "type";
+    private final String sizeCapacity = "sizeCapacity";
+    private final String b1 = "b1";
+
+    private final String howManyYears = "howManyYears";
+
+
+    @BeforeEach
+    public void Setup() {
+        String bookClassString = "oop.programs_with_multiple_classes.hard.q4." + bookUc;
+        String bookcaseClassString = "oop.programs_with_multiple_classes.hard.q4." + bookcaseUc;
+        book = new ObjectTest(bookClassString);
+        bookcase = new ObjectTest(bookcaseClassString);
+    }
+
+    /**
+     * Book tests
+     **/
+    @Test
+    public void bookClassHasCorrectFields() {
+        String missingFieldMessage = "Your " + bookLc + " class is missing a required field.";
```

Comment:
Bowen asked that we name the missing attribute, and we have been using something similar to the message below:
"The %s class is missing the %s attribute. Make sure that the class contains the field and that it is spelled correctly."